src/WLR/Types/Output.hsc

```diff
@@ -0,0 +1,431 @@
+{-# LANGUAGE EmptyDataDeriving, PatternSynonyms #-}
+
+module WLR.Types.Output where
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_output.h>
+
+import Foreign
+import Foreign.C.Types
+
+import PIXMAN.Pixman
+import WL.ServerCore
+import WL.ServerProtocol
+import WL.Utils
+import WLR.Backend
+import WLR.Render.Allocator
+import WLR.Render.DrmFormatSet
+import WLR.Render.Interface
+import WLR.Render.Renderer
+import WLR.Render.Swapchain
+import WLR.Render.Texture
+import WLR.Types.Buffer
+import WLR.Types.OutputLayer
+import WLR.Util.Addon
+import WLR.Util.Box
+
+-- A struct from <time.h>. I think this should only be handled by
+-- standard library code, so an opaque pointer should suffice.
+type TIMESPEC = ()
```

Comment:
I’d define this as a `data` marked as corresponding to `struct timespec`, I think.